atlite/wind.py

```diff
@@ -65,22 +69,52 @@ def extrapolate_wind_speed(ds, to_height, from_height=None):
         if len(heights) == 0:
             raise AssertionError("Wind speed is not in dataset")
 
-        from_height = heights[np.argmin(np.abs(heights - to_height))]
-
-    from_name = "wnd{h:0d}m".format(h=int(from_height))
-
-    # Wind speed extrapolation
-    wnd_spd = ds[from_name] * (
-        np.log(to_height / ds["roughness"]) / np.log(from_height / ds["roughness"])
-    )
-
-    wnd_spd.attrs.update(
-        {
+        if model == "without_roughness" and len(heights) > 1:
+            from_height = heights[np.argmin(np.abs(heights - to_height))]
+            heights = np.delete(heights, np.argmin(np.abs(heights - to_height)))
+            from_height2 = heights[np.argmin(np.abs(heights - to_height))]      # Finding second closest height
+            print("Calculating wind speed at hub height using 2 wind speed datasets.")
+        
+        elif model == "with_roughness" or len(heights) == 1: 
+            from_height = heights[np.argmin(np.abs(heights - to_height))]
+            model == "with_roughness"
+            print("Calculating wind speed at hub height using 1 wind speed dataset and surface roughness dataset.")
+        else:
+            raise AssertionError("model has to be either 'with_roughness' or 'without_roughness'")
+
+    # Wind speed extrapolation with the option to NOT use surface roughness, using the wind speed measured at two heights: 
+    if model == "without_roughness": 
+        print("from_height",from_height)
+        print("from_height2",from_height2)
+        from_name = "wnd{h:0d}m".format(h=int(from_height))
+        from_name2 = "wnd{h:0d}m".format(h=int(from_height2))         # definition of second wind speed (for instance at 100m)
+
+        wnd_spd = ds[from_name] * (to_height / from_height)**((1 / np.log(from_height)) * np.log(ds[from_name2] / ds[from_name]))
```

Comment:
Shouldn't the  equation be:

> wnd_spd = ds[from_name] * (to_height / from_height)**((np.log(from_height2) / np.log(from_height)) * np.log(ds[from_name2] / ds[from_name]))

Reply:
Yes, you are right. The former formulation only works when "from_height2" = 100 m and "from_height" = 10 m.

I would agree with Oskar's formulation (https://github.com/OskarSidor/atlite/commit/bc8c6228f2ad142b9550661460527a3e429d7003#r132404361):

```
wnd_spd = ds[from_name] * (to_height / from_height)**
(
np.log(ds[from_name2] / ds[from_name]) 
/ 
np.log(from_height2 / from_height)
)
```